js/app.js

```diff
@@ -0,0 +1,188 @@
+
+    // obstaculos
+    frameCounter:0,
+    obstaculos:[],
+    
+    imageInstance: undefined,
+    canvasSize: {
+        w: undefined,
+        h: undefined
+    },
+
+   
+    
+    // car!! 
+    car: {
+        pos: { x: 215, y: 570 },
+        size:{w:60,h:120},
+        image: '/images/car.png' 
+    },
+
+
+
+    init() {
+        // road
+        this.setDimensions()
+        this.setContext()
+        this.drawranguloverde()
+        this.drawRoadGris()
+        this.drawLeftBorde()
+        this.drawRightBorde()
+        this.drawLineaDis()
+        
+        // car
+
+        this.setEventHandlers()
+        this. createCar()
+       
+         // crear obstaculo
+        this.createObstaculos()
+        
+
+        this.start()
+        
+       
+
+        
+        
+    },
+
+    setDimensions() {
+        this.canvasSize = {
+            w: 500,
+            h: 700,
+        }
+    },
+
+    setContext() {
+        this.ctx = document.querySelector('#canvas').getContext('2d')
+    },
+    //   crear el fondo verde
+    drawranguloverde() {
+        this.ctx.fillStyle = 'green'
+        this.ctx.fillRect(0, 0, this.canvasSize.w, this.canvasSize.h)
+      
+    },
+
+    // crear el fondo gris
+
+    drawRoadGris() {
+        this.ctx.fillStyle = 'grey'
+        this.ctx.fillRect(50, 0, this.canvasSize.w - 100, this.canvasSize.h)
+        
+    },
+    
+
+    // left Borde
+    drawLeftBorde() {
+        this.ctx.fillStyle = 'white'
+        this.ctx.fillRect(70, 0, 10, this.canvasSize.h)
+    },
+    // right Borde
+    drawRightBorde() {
+        this.ctx.fillStyle = 'white'
+        this.ctx.fillRect(this.canvasSize.w - 70, 0, 10, this.canvasSize.h)
+    },
+    //  liena discontinua
+    drawLineaDis() {
+        this.ctx.beginPath()
+        this.ctx.moveTo(this.canvasSize.w / 2 - 5, 0)
+        this.ctx.lineTo(this.canvasSize.w / 2 - 5, this.canvasSize.h)
+        this.ctx.setLineDash([60, 20])
+        this.ctx.lineWidth = 10
+        this.ctx.strokeStyle = 'white'
+        this.ctx.stroke()
+        this.ctx.closePath()
+    },
+    
+
+
+
+
+    // car part
+   
+   
+
+
+    createCar() {
+        this.imageInstance = new Image()
+        this.imageInstance.src = this.car.image
+    },
+   
+   setEventHandlers() {
+        document.onkeydown = event => {
+            switch (event.key) {
+                case 'ArrowLeft':
+                    this.car.pos.x -= 10
+                    break;
+                case 'ArrowRight':
+                    this.car.pos.x += 10
+                    break;
+                 case 'ArrowUp':
+                    this.car.pos.y -= 10
+                    break;
+                case 'ArrowDown':
+                    this.car.pos.y += 10
+                    break;
+                
+            }
+        }
+    },
+   
+   
+//    crear obstaculos
+    createObstaculos() {
+        this.obstaculos.push(
+            new Obstaculos(this.ctx,this.canvasSize)
+        )
+        
```

Comment:
Para ver si se estaban generando obstaculos he puesto un console.log(this.obstaculos)
He comprobado que si se estaban generando pero las medidas que salian no tenian ningun sentido.